src/dolphin/unwrap/_unwrap.py

```diff
@@ -450,6 +441,20 @@ def unwrap(
             scratchdir=scratchdir,
         )
 
+        # Move the intermediate ".interp" or ".goldstein" into the scratch directory
+        if scratchdir is not None:
+            shutil.move(unwrapper_unw_filename, scratchdir)
```

Comment:
Noting that this was added to avoid having the `.filt.interp` intermediate files in the `unwrapped` folder. those should be in scratch. The tests I added were to ensure these don't stick around in `unwrapped/`